org.alloytools.alloy.core/src/main/java/edu/mit/csail/sdg/alloy4/OurDialog.java

```diff
@@ -215,10 +227,14 @@ public static File askFile(boolean isOpen, String dir, final String ext, final S
 			open.setAlwaysOnTop(true);
 			open.setMode(isOpen ? FileDialog.LOAD : FileDialog.SAVE);
 			open.setDirectory(dir);
-			if (ext.length() > 0)
+			if (exts != null && exts.length > 0)
```

Comment:
This `if (exts != null && exts.length > 0)` appears 3 times in this method.

I think either (1) `exts != null` should be a precondition (just throw exception if someone calls this method with `null`), or (2) it should be automatically sanitized and replaced with an empty array if `null`.  That way you can get rid of the first two checks in this method body, and only for the third check you will probably have to check if the array is empty.

Reply:
Look there is so much of this in the code that I think we should clean up more systematically. Not sure I want to handle that in this PR.